email-reader/src/test/java/no/sikt/nva/email/reader/handler/ScopusEmailReaderTest.java

```diff
@@ -0,0 +1,182 @@
+        validEmail = EmailGenerator.generateValidEmail();
+    }
+
+
+    @Test
+    void shouldThrowExceptionIfItCannotParseTheEmail() throws IOException {
+        var s3EventNotContainingAnEmail = createS3Event(randomString());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotContainingAnEmail, CONTEXT));
+        assertThat(exception.getMessage(), containsString(COULD_NOT_VERIFY_EMAIL));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotContainingAnEmail))));
+    }
+
+
+    @ParameterizedTest(name = "should throw exception when email sender is not from Elsevier email account")
+    @MethodSource("generateInvalidSenderEmail")
+    void shouldThrowExceptionWhenEmailSenderIsNotFromElsevierEmailAccount(String invalidEmails) throws IOException {
+        var s3EventNotFromSikt = createS3Event(invalidEmails);
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(COULD_NOT_VERIFY_EMAIL));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+    @Test
+    void shouldThrowExceptionIfEmailDoesNotHaveContentTypeMultipart() throws IOException, MimeException {
+        var s3EventNotFromSikt = createS3Event(EmailGenerator.generateNonMultipartEmail());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(COULD_NOT_PARSE_EMAIL));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+
+    @Test
+    void shouldThrowExceptionIfEmailDoesNotContainBodyWithURI() throws IOException, MimeException {
+        var s3EventNotFromSikt = createS3Event(EmailGenerator.generateEmailWithoutScopusLinks());
+        var exception = assertThrows(EmailException.class, () -> handler.handleRequest(s3EventNotFromSikt, CONTEXT));
+        assertThat(exception.getMessage(), containsString(NO_URL_PRESENT_IN_MESSAGE));
+        assertThat(exception.getObjectKey(), is(equalTo(extractObjectKey(s3EventNotFromSikt))));
+        assertThat(exception.getBucket(), is(equalTo(INPUT_BUCKET_NAME)));
+    }
+
+    @Test
+    void shouldNotDownloadCitedByUri() throws IOException {
+        var s3EventNotFromSikt = createS3Event(validEmail);
+        var actualUrl = handler.handleRequest(s3EventNotFromSikt, CONTEXT);
+        assertThat(actualUrl, not(contains(UriWrapper.fromUri("https://sccontent-scudd-delivery-prod.s3.amazonaws.com/sccontent-scudd-delivery-prod/cristin_1/2023-6-14/cristin_1_2023-6-14_ANI-CITEDBY.zip?my-query-param=should-be-preserved").getUri())));
```

Comment:
what value does this show in debug?

Reply:
Merging this test with another. so it is more clear what URL should be present and which one should not be.